File: Real-estate-market-IB190099.Services/Database/DemandMessage.cs
using System;
using System.Collections.Generic;

namespace Real_estate_market_IB190099.Services.Database;

public partial class DemandMessage
{
    public int Id { get; set; }

    public string Content { get; set; } = null!;

    public int? SenderId { get; set; }

    public int? DemandAdvertiseId { get; set; }

    public byte? IsEmployee { get; set; }

    public DateTime? Timestamp { get; set; }

    public virtual DemandAdvertise? DemandAdvertise { get; set; }

    public virtual User? Sender { get; set; }
}

File: Real-estate-market-IB190099.mobile/realestatemobile/lib/model/demand_message.dart
import 'package:realestatemobile/model/user.dart';
// ignore_for_file: non_constant_identifier_names, depend_on_referenced_packages
import 'package:json_annotation/json_annotation.dart';
import 'demand_advertise.dart';
part 'demand_message.g.dart';

@JsonSerializable()
class DemandMessage {
  int? id;
  String? content;
  User? sender;
  int? senderId;
  Object? timestamp;
  DemandAdvertise? demandAdvertise;
  int? demandAdvertiseId;
  int? isEmployee;

  DemandMessage() {}

  factory DemandMessage.fromJson(Map<String, dynamic> json) =>
      _$DemandMessageFromJson(json);

  /// Connect the generated [_$DemandMessageToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$DemandMessageToJson(this);
}
